src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -6650,12 +6650,45 @@ and TcCtorCall isNaked cenv env tpenv (overallTy: OverallTy) objTy mObjTyOpt ite
     if isInterfaceTy g objTy then
         error(Error((if superInit then FSComp.SR.tcInheritCannotBeUsedOnInterfaceType() else FSComp.SR.tcNewCannotBeUsedOnInterfaceType()), mWholeCall))
 
+    let rec getConstructorArgs expr =
+        match expr with
+        | SynExpr.Paren(expr = SynExpr.App(funcExpr = expr)) -> getConstructorArgs expr
+        | SynExpr.Paren(expr = SynExpr.Tuple(exprs = synExprs)) -> synExprs |> List.collect getConstructorArgs
+        | SynExpr.App(funcExpr = expr1; argExpr = expr2) -> [ expr1 ; expr2 ] |> List.collect getConstructorArgs
+        | SynExpr.Ident(id) -> [ id ]
+        | _ -> []
+
     match item, args with
-    | Item.CtorGroup(methodName, minfos), _ ->
+    | Item.CtorGroup(methodName, minfos), argExprs ->
         let meths = List.map (fun minfo -> minfo, None) minfos
         if isNaked && TypeFeasiblySubsumesType 0 g cenv.amap mWholeCall g.system_IDisposable_ty NoCoerce objTy then
             warning(Error(FSComp.SR.tcIDisposableTypeShouldUseNew(), mWholeCall))
 
+        let ctorArgs = argExprs |> List.collect getConstructorArgs
+        
+        if not ctorArgs.IsEmpty then
+            // Here we will have the ValRefs(Members, Properties) from F# types
+            let valRefs =
+                minfos
+                |> List.collect (fun minfo-> minfo.ApparentEnclosingTyconRef.MembersOfFSharpTyconSorted)
+                |> List.filter (fun v -> CheckFSharpAttributesForObsolete g v.Attribs)
+
+            if valRefs.IsEmpty then
+                // Here we will have the PropInfos from C# types
+                let propInfos =
+                    minfos
+                    |> List.collect (fun minfo -> GetImmediateIntrinsicPropInfosOfType (None, AccessibleFromSomeFSharpCode) g cenv.amap range0 minfo.ApparentEnclosingType)
+
+                for propInfo in propInfos do
```

Comment:
I don't like the penalty hit this would mean for every call which is not even using setter properties in a constructor.

I would much more favor iterating just over the properties which are really set.
If you search for `let finalAssignedItemSetters = finalCalledMeth.AssignedItemSetters` in CheckExpressions, you will find that data for it.

Reply:
Also prop setters can be included in methods which are not constructors, so I think TcMethodApplication would be more appropriate here (should cover both constructors as well as methods IIRC).

![image](https://github.com/dotnet/fsharp/assets/46543583/b3d16c7d-d63f-464f-bcda-d09d4dab99cd)

Reply:
> Also prop setters can be included in methods which are not constructors, so I think TcMethodApplication would be more appropriate here (should cover both constructors as well as methods IIRC).

Yeah, I'm aware of this. Needs some investigation. In this PR I wanted to focus in https://github.com/dotnet/fsharp/issues/11868, will raise a new PR

Reply:
> I don't like the penalty hit this would mean for every call which is not even using setter properties in a constructor.
> 
> I would much more favor iterating just over the properties which are really set. If you search for `let finalAssignedItemSetters = finalCalledMeth.AssignedItemSetters` in CheckExpressions, you will find that data for it.

Done. See my last commit

Reply:
> Yeah, I'm aware of this. Needs some investigation. In this PR I wanted to focus in #11868, will raise a new PR

Created a new issue to focus on this https://github.com/dotnet/fsharp/issues/16952